jerry-core/ecma/builtin-objects/ecma-builtin-json.c

```diff
@@ -923,13 +923,20 @@ static ecma_value_t ecma_builtin_json_str_helper (const ecma_value_t arg1, /**<
                                                   empty_str_p,
                                                   arg1,
                                                   false);
-  JERRY_ASSERT (ecma_is_value_true (put_comp_val));
-  ecma_free_value (put_comp_val);
-  ECMA_TRY_CATCH (str_val,
-                  ecma_builtin_json_str (empty_str_p, obj_wrapper_p, &context),
-                  ret_value);
-  ret_value = ecma_copy_value (str_val);
-  ECMA_FINALIZE (str_val);
+
+  if (ecma_is_value_true (put_comp_val))
+  {
+    ECMA_TRY_CATCH (str_val,
```

Comment:
This can be simplified. `ECMA_TRY_CATCH` is unnecessary, the following code is equal:
```
{
  ret_value = ecma_builtin_json_str (empty_str_p, obj_wrapper_p, &context);
}
```

Reply:
`ecma_copy_value` was needed because of the `ECMA_FINALIZE`. Without the `ECMA_TRY_CATCH` and `ECMA_FINALIZE` it is not needed anymore.

Reply:
Indeed. That makes everything a lot simpler.